lib/react.js

```diff
@@ -336,10 +336,16 @@ declare module react {
   declare type MaybeCleanUpFn = void | (() => void);
 
   declare export function useContext<T>(context: React$Context<T>): T;
+    
+  type NonFunctions = number | boolean | Object | string | null | void | Array<NonFunctions>;
 
-  declare export function useState<S>(
+  declare export function useState<S: NonFunctions>(
     initialState: (() => S) | S,
   ): [S, ((S => S) | S) => void];
+  
+  declare export function useState<S: Function>(
```

Comment:
`Function` is just an alias for `any`: https://flow.org/en/docs/linting/rule-reference/#toc-unclear-type